test/extensions/quic_listeners/quiche/integration/quic_http_integration_test.cc

```diff
@@ -218,5 +228,138 @@ TEST_P(QuicHttpIntegrationTest, TestDelayedConnectionTeardownTimeoutTrigger) {
             1);
 }
 
+TEST_P(QuicHttpIntegrationTest, MultipleQuicListenersWithBPF) {
+#if defined(SO_ATTACH_REUSEPORT_CBPF) && defined(__linux__)
+  concurrency_ = 8;
+  set_reuse_port_ = true;
+  initialize();
+  std::vector<IntegrationCodecClientPtr> codec_clients;
+  quic::QuicCryptoClientConfig::CachedState* cached = crypto_config_.LookupOrCreate(server_id_);
+  for (size_t i = 1; i <= concurrency_; ++i) {
+    // The BPF filter looks at the 1st word of connection id in the packet
+    // header. And currently all QUIC versions support 8 bytes connection id. So
+    // create connections with the first 4 bytes of connection id different from each
+    // other so they should be evenly distributed.
+    cached->add_server_designated_connection_id(quic::test::TestConnectionId(i << 32));
+    codec_clients.push_back(makeHttpConnection(lookupPort("http")));
+  }
+  if (GetParam() == Network::Address::IpVersion::v4) {
+    test_server_->waitForCounterEq("listener.0.0.0.0_0.downstream_cx_total", 8u);
+  } else {
+    test_server_->waitForCounterEq("listener.[__]_0.downstream_cx_total", 8u);
+  }
+  for (size_t i = 0; i < concurrency_; ++i) {
+    if (GetParam() == Network::Address::IpVersion::v4) {
+      test_server_->waitForGaugeEq(
+          fmt::format("listener.0.0.0.0_0.worker_{}.downstream_cx_active", i), 1u);
+      test_server_->waitForCounterEq(
+          fmt::format("listener.0.0.0.0_0.worker_{}.downstream_cx_total", i), 1u);
+    } else {
+      test_server_->waitForGaugeEq(fmt::format("listener.[__]_0.worker_{}.downstream_cx_active", i),
+                                   1u);
+      test_server_->waitForCounterEq(
+          fmt::format("listener.[__]_0.worker_{}.downstream_cx_total", i), 1u);
+    }
+  }
+  for (size_t i = 0; i < concurrency_; ++i) {
+    codec_clients[i]->close();
+  }
+#endif
+}
+
+TEST_P(QuicHttpIntegrationTest, MultipleQuicListenersNoBPF) {
+#ifdef __APPLE__
```

Comment:
idle thought, we do this enough I wonder if it's worth a macro like the #ifndef ENVOY_DISABLE_DEPRECATED_FEATURES to just have
TEST_P(QuicHttpIntegrationTest, DISABLE_FOR_APPLE(MultipleQuicListenersNoBPF)) 

pros: it saves test setup and teardown on the builds we don't want.  cons: the macros don't stack so we might have to one-off any which used both.  Optional - add it in if you think it's a good idea :-)